docs/smart-contracts/install/migrate-pact5.md

```diff
@@ -0,0 +1,417 @@
+Pact 5 represents a comprehensive refactoring of the Pact language and command-line interpreter that offers improvements in efficiency, performance, and ergonomics.
+The improvements in Pact 5 ensure lower transaction costs, faster module loading, and enhanced testing and debugging capabilities.
+To take advantage of these improvements, you need to update your development environment and smart contracts to use Pact 5.
+This article covers the basics of upgrading to Pact 5 and details some of the breaking changes that might require you to update and redeploy existing contracts or fix potential coding errors.
+
+## Preparing to migrate
+
+Ideally, you should test any smart contracts you've written by installing Pact 5 locally before Pact 5 is released on the Kadena main production network.
+If you install Pact 5 locally in your development environment, you can run all of your existing contract tests you've defined in REPL files using the Pact 5 binary to verify that they work as expected.
+For information about installing the Pact 5 binary locally, see [/smart-contracts/install].
+
+You can also test existing smart contracts by deploying them on the Kadena test network.
+You can deploy contracts for testing with Pact 5 on the Kadena `testnet05` network, if you deploy before 7 February 2025.
+You can connect to the `testnet05` network by using the API node available at `api.testnet05.chainweb.com`.
+The `testnet05` network is a separate test network where Pact 5 runs on Chainweb nodes before being deployed on the `testnet04` network and the Kadena `mainnet01` network.
+Note that the `testnet05` network will be decommissioned and removed from service when the `testnet04` network is upgraded to use Pact 5.
+
+Pact 5 will also be deployed by default on the `development` network that you can run locally beginning with the `chainweb-node` 2.27 release. 
+After `chainweb-node` 2.27 is released, you can start a local development node to build and test contracts using Pact 5 by default.
+
+## After upgrading to Pact 5
+
+Most smart contracts written in Pact 4, or earlier, should be compatible with Pact 5 and continue to operate as expected without any issues. 
+However, some of the bug fixes that were found in earlier versions of Pact and corrected in the Pact 5 implementation might cause unexpected behavior in your contracts or require some type of modification to the contract code.
+
+You can find details about new features and the types of issues you might encounter in the remainder of this document.
+
+At a minimum, you should expect to redeploy existing contracts on the Kadena main production network after Pact 5 is released on the Kadena `mainnet01` network.
+Pact 5 includes a new built-in function for redeploying contracts, so you can take advantage of reduced transaction costs and faster transaction execution with minimal migration steps.
```

Comment:
```suggestion
Redeploying existing contracts is not required to use Pact 5, but will decrease gas costs, because the newly deployed version will use the Pact 5 on-disk format. Pact 5 includes a new built-in function for doing this easily.
```